enhancements/installer/component-selection.md

```diff
@@ -94,233 +95,183 @@ N/A - no api extensions are being introduced in this proposal
-were not explicitly requested by spec, but got pulled in due to the inclusionDefault, or due to a resource
-moving between capabilities(see step 5 below).
 
+`enabledCapabilities` will often align with the requested `spec.capabilities`.
+But in some situations, `enabledCapabilities` may extend the requested capability set with additional entries, as discussed below.
 
+Note: The CVO must wait until the ClusterVersion resource exists (created by the installer) before creating
+any filterable resource (a resource with a capability annotation) to ensure there is no race condition in which
+the CVO starts creating resources that should have been filtered, before it has the filter list.
 
-4) Admin can remove an item from the excluded annotations list, but they cannot add an item to it.  If an
-item is removed, the CVO will apply the previously filtered(skipped) resources to the cluster on the next reconciliation.
-Adding an item to the filtered list is not supported because it requires the component be removed from the
-running cluster which has more significant implications for how all traces of the component are removed.  The
-CVO should reject admin edits/updates that add new entries to the exclusion list.  Again this requires the
-introduction of a validating webhook for the ClusterVersion resource.
+##### Capabilities can be installed
 
-Similarly, the `inclusionDefault` cannot be changed from `include` to `exclude` after the cluster install is complete.
+Admins can adjust `spec.capabilities` to request additional capabilities, and the CVO will extend `enabledCapabilities` and begin reconciling [the newly-enabled manifests](#manifest-annotations).
 
-The currently configured capabilities settings for the CVO should be recorded in telemeter so we can understand
-the configuration of a given cluster.
+##### Capabilities cannot be uninstalled
 
-In the future, we might allow specific APIs to be disabled, such as the build api.  This could be done by
-defining additional capability keywords that can be put into the install-config field being defined here,
-which would drive the creation of cluster config that the apiserver+controllers used to disable that particular
-api.
+Admins can adjust `spec.capabilities` to stop requesting a capability, but the CVO will refuse, continue to hold that capability in `enabledCapabilities`, and continue to reconcile [the associated manifests](#manifest-annotations).
+When the requested `spec.capabilities` diverge from `enabledCapabilities`, the CVO will set `ImplicitlyEnabledCapabilities=True` in ClusterVersion's `status.conditions` with a message pointing out the divergence.
+
+Removing installed components is complicated, and we are deferring that for now.
+
+We will [consider][admission-webhook] adding an admission webhook to improve the UX for admins who attempt to remove an enabled capability, by replacing "notice `ImplicitlyEnabledCapabilities=True`" with "have your PATCH fail with a 400".
```

Comment:
> so if you want to see the list of implicitly enabled caps on your cluster, you have to look at the message associated w/ the condition?

Yes.

>  (that's ok, I guess, but i missed something because i had thought it was going to be an explicit status field that listed them)

If we'd decided that, I didn't get it captured in openshift/api@5b82635ef101e7c10b5fcbbcfb81315bb7a0da20.  Can you point me at discussion?  It seems like one of the things where we can look at `ImplicitlyEnabledCapabilities=True` prevalence and use that to inform a decision about whether the structured status would be impactful enough to bother.

Reply:
sorry i can't point to a discussion (in that i can't even begin to think what that slack search string would be, nevermind whether there actually was one or i am thinking of something else/misunderstood the discussion).  As you say, we can add it later, so we can go with this and see if we get complaints.